### ISSUE TITLE
NavigationAgent/NavigationAgent2D node does not display configuration warnings

### ISSUE THREAD

**Issue by Quantum-Cucumber** [godotengine/godot#72368]
### Godot version

3.5.1.stable

### System information

Ubuntu 18.04.6 x86_64

### Issue description

Attaching a simple script such as
```gdscript
tool
extends Node

func _get_configuration_warning() -> String:
    return "Test"
```
To either a NavigationAgent or NavigationAgent2D node does not produce a warning in the editor node tree. This script works just fine with a variety of other nodes.

### Steps to reproduce

1. Add a Node2D/Spatial inheriting node (optional)
2. Add a NavigationAgent2D/NavigationAgent node as a child
3. Attach a script that overrides `_get_configuration_warning()` and returns a non-empty string

### Minimal reproduction project

[config_warning_test.zip](https://github.com/godotengine/godot/files/10534768/config_warning_test.zip)

**Comment by smix8** [comment]
This is not specific to NavigationAgent or navigation.

NavigationAgent is just one of many more Godot 3.x nodes with the same issue that they do not propagate warnings from any parent class.

This was fixed in Godot 4 by https://github.com/godotengine/godot/pull/38743 but was not backported.

**Comment by akien-mga** [comment]
Fixed by #72386.
